src/libs/Pusher/pusher.js

```diff
@@ -358,6 +359,20 @@ function reconnect() {
     socket.connect();
 }
 
+/**
+ * Combines channel parts and adds Pusher suffix when necessary
+ *
+ * @param {*} parts
+ *
+ * @returns {String}
+ */
+function getChannelName(...parts) {
+    return _.chain([...parts, CONFIG.PUSHER.SUFFIX])
+        .filter(Boolean)
+        .join('-')
+        .value();
```

Comment:
Perhaps this should be named to `getPrivateChannelName` because I saw an instance that does not use the Pusher suffix (for the policy editor channel)

Or maybe we could try to make the logic "smarter" and try to append suffix only if the `parts` contain the word `"private"`?

---

Regarding the dash:  
Onyx collection keys include the `-`inside the key name, perhaps you'd like the same here, though filtering out empty keys would require more logic

Reply:
I like the idea of this being `getPrivateChannelName()` 👍 

> Onyx collection keys include the -inside the key name, perhaps you'd like the same here, though filtering out empty keys would require more logic

I'm not really following this and what changes it would imply. What do the parts in the pusher channel name have to do with Onyx collection keys?

Reply:
Hm, I actually don't particularly like having the dash automatically derived in this code. I think it makes it trickier to keep the channels names in PHP in sync with the channel names in JS. I would propose that this function is more like this:

```
function getChannelNameWithSuffix(name) {
    return `${name}-${CONFIG.PUSHER.SUFFIX}`;
}
```

Reply:
> I'm not really following this and what changes it would imply. What do the parts in the pusher channel name have to do with Onyx collection keys?

It's only about dashes and concatenation, do we have a preference to include the dash in the key (like in Onyx keys) or not?

#### Onyx keys
https://github.com/Expensify/App/blob/d1f3643b7254c2772176189d37dbf62baedd9e33/src/ONYXKEYS.js#L99-L103

#### Usage (no dash in concatenation)
```js
Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, actionsToRemove);
```

Following that pattern we can name Pusher channels like so:

```js
    PUSHER_CHANNEL: {
        USER: 'private-user-accountID-',
        REPORT: 'private-report-reportID-',
        POLICY_EDITOR: 'public-policyEditor-',
    },
```

Reply:
> Hm, I actually don't particularly like having the dash automatically derived in this code. I think it makes it trickier to keep the channels names in PHP in sync with the channel names in JS.

In that case maybe extracting names to CONST and a `getName` function is only getting in the way, can we stick with this?

```js
const channelName = `private-user-accountID-${accountID}-${CONFIG.PUSHER.SUFFIX}`
Pusher.subscribe(channelName);
```

I'll drop these changes and perhaps we can open a separate refactoring PR if needed